doc/building_picoquic.md

```diff
@@ -0,0 +1,84 @@
+
+TODO: Current issues and ongoing work
+
+There is a current issue trying to enable the use of `mbedtls` instead of `openssl`.
+
+## Targets
+
+Targets are anything that is defined as an executable in `CMakeLists.txt`.
+Available targets:
+
+- picoquicdemo
+- picolog_t
+- picoquic_ct
+- picohttp_ct
+- pico_baton
+- picoquic_sample
+- thread_test
+
+All of these targets are built when the `make .` or `cmake --build .` commands are used to build the project. After which the test program `picoquic_ct` can be used to verify the port.
+
+`picoquicdemo` ([found in picoquicfirst/picoquicdemo.c](../picoquicfirst/picoquicdemo.c)) and `picoquic_sample` ([sample documentation](../sample/README.md)) are a good starting point for developing an application on top of this QUIC implementation.
+
+
+## (Re)Building a Single Target
+
+Targets defined in the `CMakeLists.txt`, listed above for convenience, can be build individually using cmake's `--target` option. The command `cmake --build . --target <target_name>` can be used to build a specific target and all it's dependcies. For example `cmake --build . --target picoquic_sample` will build the sample file transfer application in [sample/](../sample/README.md) as defined in the `CMakeLists.txt`. All the required components will also be built. This works if the command is issued in the top-level directory of the repository. This is useful for recompiling test programs and programs like `picoquic_sample` and `picoquicdemo` if they have been modified. Therefore reducing build times when making small changes.
+
+This also works if `picoquic` has not been built as a whole before.
+
+## Picoquic as a Dependency
+
+When using CMake [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) can be used to fetch the `picoquic` repository from github.
+A snipped that does this can be seen in the [quicrq](https://github.com/Quicr/quicrq) directory in the `dependencies` subdirectory and is shown below.
+
+```cmake
+# Enable fetching content
+include(FetchContent)
+
+# Fetch the picoquic library, which brings in picotls with it
+FetchContent_Declare(
+    picoquic
+    GIT_REPOSITORY  https://github.com/private-octopus/picoquic.git
+    GIT_TAG         master
+)
+
+# Set the option to force picoquic to fetch the picotls
+set(PICOQUIC_FETCH_PTLS ON)
+
+# Make dependencies available
+FetchContent_MakeAvailable(picoquic)
+```
+
+FetchContent can be used to fetch a specific release or tag, if a certain version or release is desired.
+This can be used as fine grain as choosing a single commit, be specifying the commit hash.
+
+In the main `CMakeLists.txt` the subdirectory can be added using the `add_subdirectory(<directory_name>)` directive.
+
+```cmake
+# Add the subdirectory dependencies, where dependencies/CMakeLists.txt looks as above
+add_subdirectory(dependencies)
+
+# check if the package was loaded/is available and stop the build if it was not
+find_package(picoquic REQUIRED)
+```
+
```

Comment:
I've been testing this in [quicdoq](https://github.com/private-octopus/quicdoq) and got it to work when the fetching snipped was in the top-level directory, but it breaks if it is put in a subdirectory (e.g. dependencies) and added via `add_subdirectory(dependencies)`. I have looked at [quicrq](https://github.com/Quicr/quicrq) since it is fetched in a subdirectory there, but I still can get it to find the fetched content.

It works if I fetchContent in the subdirectory and then configure the path with `ccmake` though.